access/types/forms.py

```diff
@@ -587,6 +600,10 @@ def grade_field(self, i, configuration):
                         break
             if add:
                 hints.append(new_hint)
+        # Show matching parts as a feedback, if subdiff is set
+        if methods[0] == 'subdiff' and solution:
```

Comment:
I think it would make sense to generate the subdiff feedback hints in the `grade_text` method since the method returns the hints list anyway. The `grade_text` method would then not need to return any new values.